applications/GeoMechanicsApplication/custom_elements/geo_steady_state_Pw_piping_element.h

```diff
@@ -0,0 +1,297 @@
+                << std::endl;
+        }
+    }
+
+    void CheckProperties() const
+    {
+        // typical material parameters check, this should be in the check of the constitutive law.
+        // possibly check PIPE_HEIGHT, CROSS_SECTION == 1.0
+        CheckProperty(DENSITY_WATER);
+        CheckProperty(DYNAMIC_VISCOSITY);
+        CheckProperty(PIPE_HEIGHT);
+    }
+
+    void CheckProperty(const Kratos::Variable<double>& rVariable) const
+    {
+        KRATOS_ERROR_IF_NOT(GetProperties().Has(rVariable))
+            << rVariable.Name() << " does not exist in the properties of element " << Id() << std::endl;
+        KRATOS_ERROR_IF(GetProperties()[rVariable] < 0.0)
+            << rVariable.Name() << " (" << GetProperties()[rVariable]
+            << ") has an invalid value at element " << Id() << std::endl;
+    }
+
+    void CheckForNonZeroZCoordinateIn2D() const
+    {
+        if constexpr (TDim == 2) {
+            const auto& r_geometry = GetGeometry();
+            auto        pos        = std::find_if(r_geometry.begin(), r_geometry.end(),
+                                                  [](const auto& node) { return node.Z() != 0.0; });
+            KRATOS_ERROR_IF_NOT(pos == r_geometry.end())
+                << "Node with non-zero Z coordinate found. Id: " << pos->Id() << std::endl;
+        }
+    }
+
+    static void AddContributionsToLhsMatrix(MatrixType& rLeftHandSideMatrix,
+                                            const BoundedMatrix<double, TNumNodes, TNumNodes>& rPermeabilityMatrix)
+    {
+        rLeftHandSideMatrix = rPermeabilityMatrix;
+    }
+
+    void AddContributionsToRhsVector(VectorType& rRightHandSideVector,
+                                     const BoundedMatrix<double, TNumNodes, TNumNodes>& rPermeabilityMatrix,
+                                     const array_1d<double, TNumNodes>& rFluidBodyVector) const
+    {
+        const auto permeability_vector =
+            array_1d<double, TNumNodes>{-prod(rPermeabilityMatrix, GetNodalValuesOf(WATER_PRESSURE))};
+        rRightHandSideVector = permeability_vector + rFluidBodyVector;
+    }
+
+    Vector CalculateIntegrationCoefficients(const Vector& rDetJContainer) const
+    {
+        const auto& r_integration_points = GetGeometry().IntegrationPoints(GetIntegrationMethod());
+
+        auto result = Vector{r_integration_points.size()};
+        // all governed by PIPE_HEIGHT and element length so without CROSS_AREA
+        std::transform(r_integration_points.begin(), r_integration_points.end(), rDetJContainer.begin(),
+                       result.begin(), [](const auto& rIntegrationPoint, const auto& rDetJ) {
+            return rIntegrationPoint.Weight() * rDetJ;
+        });
+        return result;
+    }
+
+    Matrix FillPermeabilityMatrix(double pipe_height) const
+    {
+        Matrix constitutive_matrix(1, 1);
+        constitutive_matrix(0, 0) = std::pow(pipe_height, 3) / 12.0;
+        return constitutive_matrix;
```

Comment:
Perhaps we can write this as a one-liner:
```suggestion
        return ScalarMatrix{1, 1, std::pow(pipe_height, 3) / 12.0};
```

Reply:
Short and clear, thank you. Done